smart_open/smart_open_lib.py

```diff
@@ -744,6 +744,10 @@ def __exit__(self, type, value, traceback):
         self.close()
 
 
+def s3_iter_bucket_process_key_with_args(args):
```

Comment:
I'd prefer `kwargs` -- too easy to make a mistake in the order of arguments otherwise. Better use named arguments (beyond the mandatory `key`, which can be positional).